### DISCUSSION TITLE
Cannot build server

### DISCUSSION THREAD

**Discussion by rcontesti** [ggml-org/llama.cpp#7941] [Q&A] [answered]
Hi, I have been trying to build the server. As silly as it sounds, I'm not sure what is the right folder. In any case I have tried following https://github.com/ggerganov/llama.cpp/blob/master/examples/server/README.md both in the main folder as well as in /examples/server.

When I run:
```
make llama-server
```
I get 
```
make: *** No rule to make target 'llama-server'.  Stop.
```
In the main folder, I'm getting:

```
I ccache not found. Consider installing it for faster compilation.
I llama.cpp build info: 
I UNAME_S:   Linux
I UNAME_P:   x86_64
I UNAME_M:   x86_64
I CFLAGS:    -I. -Icommon -D_XOPEN_SOURCE=600 -D_GNU_SOURCE -DNDEBUG -DGGML_USE_OPENMP -DGGML_USE_LLAMAFILE  -std=c11   -fPIC -O3 -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wshadow -Wstrict-prototypes -Wpointer-arith -Wmissing-prototypes -Werror=implicit-int -Werror=implicit-function-declaration -pthread -march=native -mtune=native -fopenmp -Wdouble-promotion 
I CXXFLAGS:  -std=c++11 -fPIC -O3 -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wmissing-declarations -Wmissing-noreturn -pthread -fopenmp  -march=native -mtune=native -Wno-array-bounds -Wno-format-truncation -Wextra-semi -I. -Icommon -D_XOPEN_SOURCE=600 -D_GNU_SOURCE -DNDEBUG -DGGML_USE_OPENMP -DGGML_USE_LLAMAFILE 
I NVCCFLAGS: -std=c++11 -O3 
I LDFLAGS:    
I CC:        cc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
I CXX:       c++ (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0

make: *** No rule to make target 'llama-server'.  Stop.
```
Other than that, I have been able to use llama.cpp without issues. It is just a matter of the server. Thank you.

**Comment by supportend** [comment, marked as answer]
I have the executable in my main folder, perhaps because i compile llama.cpp with examples, LLAMA_BUILD_EXAMPLES=1:

`make CC=/usr/bin/gcc-13 CXX=/usr/bin/g++-13 LLAMA_OPENBLAS=1 LLAMA_BUILD_EXAMPLES=1 LLAMA_BUILD_LIBS=1 LLAMA_FAST=1`

Remove the strings, when you don't need them. I use a newer compiler, OpenBlas, ...
On my system with newest llama.cpp source, make llama-server works too.

But i think, "server" is there without building examples.

**Reply by rcontesti** [reply]
> I have the executable in my main folder, perhaps because i compile llama.cpp with examples, LLAMA_BUILD_EXAMPLES=1:
> 
> `make CC=/usr/bin/gcc-13 CXX=/usr/bin/g++-13 LLAMA_OPENBLAS=1 LLAMA_BUILD_EXAMPLES=1 LLAMA_BUILD_LIBS=1 LLAMA_FAST=1`
> 
> Remove the strings, when you don't need them. I use a newer compiler, OpenBlas, ... On my system with newest llama.cpp source, make llama-server works too.
> 
> But i think, "server" is there without building examples.

Thanks for your quick reply. Not sure I follow. Are you able to build the server with the command that you show?

**Reply by rcontesti** [reply]
Just found my error. Everything works find. I was just using an old command. Instead of running ```llama-server``` I should run ```server```
